src/mds/MDSMap.cc

```diff
@@ -462,6 +459,9 @@ void MDSMap::mds_info_t::decode(bufferlist::iterator& bl)
   if (struct_v >= 6) {
     ::decode(standby_for_fscid, bl);
   }
+  if (struct_v >= 7) {
+    ::decode(standby_replay, bl);
+  }
```

Comment:
I think we should be trying to infer standby_replay state of the MDS on decode, if it's an older encoding. Otherwise it can lose the state, right, and I don't think it'll get it re-granted automatically unless maybe the MDS requests it to go back (not sure).